### DISCUSSION TITLE
Weird typing behavior changes from Prisma 4.15.0 to 4.16.x

### DISCUSSION THREAD

**Discussion by kazazor** [prisma/prisma#22525] [Q&A]
### Question

I'm in the process of upgrading from Prisma 3 to 4 (5 will come later) and everything looks good up until version 4.15.0, but then weird typing behavior starts at 4.16.x.

# A problem that is happening only on 4.16.x but didn't on 4.15.0

```js
model Batch {
  id               Int           @id @default(autoincrement())
  campaignId       String        @map("campaign_id") @db.VarChar(10)
  campaign         Campaign      @relation(fields: [campaignId], references: [id])
  members          BatchMember[]
  createdAt        DateTime      @default(now()) @map("created_at")
  updatedAt        DateTime      @updatedAt @map("updated_at")
  sequence         Int
  filters          Json?
  videoSubmissions Int?          @map("video_submissions")
  status           BatchStatus   @default(CREATED)

  emailCampaigns BatchEmailCampaign[]

  @@map("batch")
}
```

On 4.15.0:
<img width="665" alt="Screenshot 2023-12-29 at 11 34 26" src="https://github.com/prisma/prisma/assets/1577655/000de066-b7d0-4390-bc56-9a628a59c626">

It is recognized correctly as `number`.

And then on 4.16.0:
<img width="675" alt="Screenshot 2023-12-29 at 11 29 02" src="https://github.com/prisma/prisma/assets/1577655/668a1ae2-5fa8-4ef6-bfea-d5a67f178b2d">

And here it's `any`.

I don't see anything special in the release notes of 4.16.0 that could have caused that and to be honest I can't find the reason why this is happening.
I even removed the `_count` part to see if it had anything related to the `filteredRelationCount` GA. But still even selecting only the `id` field with `true` returns `any` instead of `number`.

What am I missing here?

<hr>

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: macOS 14.2.1
- Database: Postgres 13.x
- Node.js version: 18.18.2
- Prisma `4.16.2`

**Comment by ludralph** [comment]
Hi @kazazor 👋 

Thank you for raising this question.

Did you please try the patch release [4.16.1](https://github.com/prisma/prisma/releases/tag/4.16.1) or [4.16.2](https://github.com/prisma/prisma/releases/tag/4.16.2) as there were some bug fixes for type issues?

**Reply by kazazor** [reply]
Tried it already, same thing

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
